zkvm/src/prover.rs

```diff
@@ -57,6 +58,21 @@ impl<'a, 'b> Delegate<r1cs::Prover<'a, 'b>> for Prover<'a, 'b> {
         Ok(run.program.pop_front())
     }
 
+    fn new_run(&self, program: &[u8]) -> Result<Self::RunType, VMError> {
```

Comment:
I think the program must be Data item, so for the prover it is converted to Data::Program (there is likely already a method for it), because that’s what allows having witnesses during vm execution.